src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFile.cs

```diff
@@ -40,13 +40,10 @@ public LockFileTarget GetTarget(NuGetFramework framework, string runtimeIdentifi
 
         public LockFileTarget GetTarget(string frameworkAlias, string runtimeIdentifier)
         {
-            var framework = PackageSpec.TargetFrameworks.FirstOrDefault(tfi => tfi.TargetAlias.Equals(frameworkAlias, StringComparison.OrdinalIgnoreCase))?.FrameworkName;
-
-            if (framework != null)
-            {
-                return GetTarget(framework, runtimeIdentifier);
-            }
-            return null;
+            return Targets.FirstOrDefault(t =>
+                t.TargetAlias.Equals(frameworkAlias) &&
+                ((string.IsNullOrEmpty(runtimeIdentifier) && string.IsNullOrEmpty(t.RuntimeIdentifier) ||
```

Comment:
Is the check for NullOrEmpty redudant here?  If one is null or empty and the other isn't, then String.Equals() returns false.  If they're both empty, then I think the logic is the same as what you have.

This entire condition statement was hard to understand.